core/src/main/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpServiceFactory.java

```diff
@@ -290,6 +340,14 @@ private static AnnotatedHttpServiceElement create(String pathPrefix, Object obje
                                   "invalid HTTP status code: %s (expected: >= 0)", statusCode);
                     return HttpStatus.valueOf(statusCode);
                 });
+        final HttpHeaders defaultHeaders = HttpHeaders.of(defaultResponseStatus(defaultResponseStatus, method));
+
+        final Consumer<AdditionalHeader> consumer = header -> {
+            defaultHeaders.set(HttpHeaderNames.of(header.name()), header.value());
+        };
+
+        findRepeatableAnnotations(clazz, AdditionalHeader.class).forEach(consumer);
+        findRepeatableAnnotations(method, AdditionalHeader.class).forEach(consumer);
```

Comment:
Could we log a warning message when there are `@AdditionalHeader` annotations with a duplicate header name at the same level? e.g.

```java
@AdditionalHeader(name = "a", value = "b")
@AdditionalHeader(name = "c", value = "d")
@AdditionalHeader(name = "c", value = "e") // WARN
public class Foo {
    @AdditionalHeader(name = "a", value = "f") // DO NOT WARN
    @AdditionalHeader(name = "g", value = "h")
    @AdditionalHeader(name = "g", value = "i") // WARN
    public void method() {}
}
```